packages/web/src/hooks/common/use-esc-close-modal.tsx

```diff
@@ -0,0 +1,19 @@
+import { useEffect } from "react";
+
+function useEscCloseModal(callback: () => void) {
+  const handleEsc = (event: KeyboardEvent) => {
+    if (event.key === "Escape") {
+      callback();
+    }
+  };
+
+  useEffect(() => {
+    window.addEventListener("keydown", handleEsc);
+
+    return () => {
+      window.addEventListener("keydown", handleEsc);
+    };
+  });
```

Comment:
If you don't add `[]`, it will run multiple times. Please check.

Reply:
done